operator/v1/types_ingress.go

```diff
@@ -1346,6 +1346,75 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// headersManipulation specifies options for performing action on headers.
+	// Currently one either perform a Set or Delete operation on HTTP Response headers.
+	// For Setting Headers you can provide a value for the header whereas for deleting one needs to provide header name.
+	// +kubebuilder:validation:Optional
+	HeadersManipulation IngressControllerHTTPHeadersManipulation `json:"headersManipulation,omitempty"`
+}
+
+// IngressControllerHeadersManipulation defines whether we want to set/delete Response or Request Headers.
+// Currently set/delete of HTTP Response Headers is available only.
+type IngressControllerHTTPHeadersManipulation struct {
+	// Response is a list of HTTP Headers which either need to set or deleted.
```

Comment:
> Is this field optional or required? It needs a tag.
> 
> I think if we make this optional, we can improve the discoverability, by removing the omitempty tag
> 
> I'd also like to see a bit more detail about the responses that are being referred to here, what is the response from for example? Perhaps we can also just say, "is a list of HTTP headers and actions to take upon them" rather than being specific about set and delete at this level

OK will make it optional and add a godoc text. It is for HTTP response headers.

Reply:
but now we are actually considering to get HTTP Requests also.